examples/generative/corrdiff_plus_plus/README.md

```diff
@@ -0,0 +1,149 @@
+```
+where `</path/to/encoder/model>` and `</path/to/diffusion/model>` should point to the encoder and diffusion model checkpoints, respectively, and `</path/to/output/file>` indicates the output NetCDF4 file.
+
+You can open the output file with e.g. the Python NetCDF4 library. The inputs are saved in the `input` group of the file, the ground truth data in the `truth` group, and the CorrDiff prediction in the `prediction` group.
+
+## Configs
+
+The `conf` directory contains the configuration files for the model, data,
+training, etc. The configs are given in YAML format and use the `omegaconf`
+library to manage them. Several example configs are given for training
+different models that are encoder only, flow matching with pretrained encoder, and encoder with diffusion
+models.
+The default configs are set to train the encoder with diffusion model.
+To train the other models, please adjust `conf/config_training.yaml`
+according to the comments. Alternatively, you can create a new config file
+and specify it using the `--config-name` option.
+
+
+## Dataset & Datapipe
+
+In this example, CorrDiff training is demonstrated on the Taiwan dataset,
+conditioned on the [ERA5 dataset](https://www.ecmwf.int/en/forecasts/dataset/ecmwf-reanalysis-v5).
+We have made this dataset available for non-commercial use under the
+[CC BY-NC-ND 4.0 license](https://creativecommons.org/licenses/by-nc-nd/4.0/legalcode.en)
+and can be downloaded from [https://catalog.ngc.nvidia.com/orgs/nvidia/teams/modulus/resources/modulus_datasets_cwa](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/modulus/resources/modulus_datasets_cwa)
+by `ngc registry resource download-version "nvidia/modulus/modulus_datasets_cwa:v1"`.
+The datapipe in this example is tailored specifically for the Taiwan dataset.
+A light-weight datapipe for the HRRR dataset is also available and can be used
+with the CorrDiff++ model.
+For other datasets, you will need to create a custom datapipe.
+You can use the lightweight HRRR datapipe as a starting point for developing your new one.
+
+
+
+### Sampling and Model Evaluation
+
+Model evaluation is split into two components. `generate.py` creates a netCDF file
+for the generated outputs, and `score_samples.py` computes deterministic and probablistic
+scores.
+
+To generate samples and save output in a netCDF file, run:
+
+```bash
+python generate.py
+```
+This will use the base configs specified in the `conf/config_generate.yaml` file.
+
+Next, to score the generated samples, run:
+
+```bash
+python score_samples.py path=<PATH_TO_NC_FILE> output=<OUTPUT_FILE>
+```
+
+Some legacy plotting scripts are also available in the `inference` directory.
+You can also bring your checkpoints to [earth2studio]<https://github.com/NVIDIA/earth2studio>
```

Comment:
What is meant by legacy plotting scripts here? If they don't work with the output generated by the scoring or generation scripts, I suggest just removing them. Also, the earth2studio link here doesn't render properly, FYI.